### PR TITLE
Fixes for FA2 adjustSizes functionality

### PR DESCRIPTION
**Pull request by apitts**
As per this discussion: https://github.com/jacomyal/sigma.js/issues/720

@Yomguithereal please note:

1. This line is clearly a hack: `prefix = this.sigInst.renderers[0].options.prefix`. I tried using rendererIndex as for the noverlap layout instead of this hardcoded zero but for some reason FA2 didn't like that approach...you may have more luck. Whenever a distance computation involving node size is involved I think we should be using rendered sizes and co-ordinates however.  
2. The computation of forces now seems in line with the Gephi algorithm here: https://github.com/gephi/gephi/tree/master/modules/LayoutPlugin/src/main/java/org/gephi/layout/plugin/forceAtlas2. However, the application of forces seems very different...I can see the parallels but there are enough differences to make it hard to follow directly. The change I made regarding maxForce seems to bring it a bit more in line with Gephi and fixed a bug for me...but, like I said, it's different...
3. Let me know if adjustSizes with Barnes-Hut on is working for you...I'm still having a bit of an issue with that.

### CODE REVIEW

**Comment by Yomguithereal** [comment]
Hello @apitts. Can you ping me later in the week so I can review your PR correctly please?

**Comment by apitts** [comment]
Sure - will ping you here. Please feel free to email me (see profile) if
you would like to discuss live at any stage.

**Comment by Yomguithereal** [comment]
Can I ask you to split this PR into two different pieces @apitts? One with the prefix thing and one with the math fixes?

The reason is I can't merge the prefix thing right now because it would not be fit for production use and we have to discuss it a bit more.

**Comment by Yomguithereal** [comment]
Concerning the last part with nodespeed and all, I don't clearly remember if there were evolutions made by the algorithm. Upon which version of the Java algorithm are you basing yourself?

**Comment by apitts** [comment]
@Yomguithereal I can remove the prefix part...my suspicion (and experience) though is that the adjustSizes setting won't work unless actual node sizes are used.

I used the latest version: 
https://github.com/gephi/gephi/tree/master/modules/LayoutPlugin/src/main/java/org/gephi/layout/plugin/forceAtlas2

Your are quite right of course...the differences could well be due to this commit (I didn't look at the history at the time):
https://github.com/gephi/gephi/commit/7b45ef33e98a856674364dcb5f69129778c4ccad#diff-5593be15f185631edfdee023e4b56ca7

**Comment by apitts** [comment]
@Yomguithereal I have removed the prefix as requested. Understand that there probably needs to be some internal discussion about what to do re the prefix issue...will hold of on adding a pull request for it until that discussion has been had.

**Comment by Yomguithereal** [comment]
Thanks @apitts. I'll have to check with @jacomyma the part about the node's speed before merging though. I can't remember if the code evolved from the Java version at one point or not.
